pkg/k8shandler/collection.go

```diff
@@ -3,31 +3,56 @@ package k8shandler
 import (
 	"fmt"
 	"github.com/openshift/cluster-logging-operator/pkg/utils"
+	"github.com/sirupsen/logrus"
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
+	"reflect"
 
 	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
 	sdk "github.com/operator-framework/operator-sdk/pkg/sdk"
+	apps "k8s.io/api/apps/v1"
 )
 
 func CreateOrUpdateCollection(logging *logging.ClusterLogging) (err error) {
-	if err = createOrUpdateFluentdServiceAccount(logging); err != nil {
-		return
-	}
 
-	if err = createOrUpdateCollectionPriorityClass(logging); err != nil {
-		return
-	}
+	if logging.Spec.Collection.Type == "fluentd" {
+		if err = createOrUpdateFluentdServiceAccount(logging); err != nil {
+			return
+		}
 
-	if err = createOrUpdateFluentdConfigMap(logging); err != nil {
-		return
-	}
+		if err = createOrUpdateCollectionPriorityClass(logging); err != nil {
+			return
```

Comment:
What happens if the operation fails?

Reply:
It will try to recreate it again the next time it goes through. The error handling for this should be  handled within the function `createOrUpdateCollectionPriorityClass`